### PR TITLE
[1.1.x] CAL-436 Add CMETAA TRE Support

### PR DESCRIPTION
**Pull request by derekwilhelm**
#### What does this PR do?

Add support for the NITF CMETAA TRE.

#### Who is reviewing it? 
@ryeats 
@dcruver 

#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)

@clockard
@jlcsmith 

#### How should this be tested?
This needs to be built with this imaging-nitf PR: [IMG-244 PR#190](https://github.com/codice/imaging-nitf/pull/190)
Ingest a NITF with a CMETAA TRE
Verify that the metacard has the "ext.nitf.cmetaa.*" attributes

#### What are the relevant tickets?

[CAL-436](https://codice.atlassian.net/browse/CAL-436)
[IMG-244](https://codice.atlassian.net/browse/IMG-244)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/complex/CmetaaAttribute.java [hunk 1]
```diff
@@ -0,0 +1,1764 @@
...
+          "RD_MODE",
+          tre -> TreUtility.getTreValue(tre, "RD_MODE"),
+          BasicTypes.STRING_TYPE);
+
+  public static final CmetaaAttribute RD_PATCH_NO_ATTRIBUTE =
+      new CmetaaAttribute(
+          RD_PATCH_NO,
+          "RD_PATCH_NO",
+          tre -> TreUtility.getTreValue(tre, "RD_PATCH_NO"),
+          BasicTypes.STRING_TYPE);
```
Comment (jlcsmith) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/complex/CmetaaAttribute.java]: ❓ should this be an integer type?

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/complex/CmetaaAttribute.java [hunk 2]
```diff
@@ -0,0 +1,1764 @@
...
+          BasicTypes.FLOAT_TYPE);
+
+  public static final CmetaaAttribute CMPLX_LINLOG_TP_ATTRIBUTE =
+      new CmetaaAttribute(
+          CMPLX_LINLOG_TP,
+          "CMPLX_LINLOG_TP",
+          tre -> TreUtility.convertToInteger(tre, "CMPLX_LINLOG_TP"),
+          BasicTypes.INTEGER_TYPE);
+
+  public static final CmetaaAttribute CMPLX_PHASE_QUANT_FLAG_ATTRIBUTE =
```
Comment (jlcsmith) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/complex/CmetaaAttribute.java]: ❓ I am not seeing this attribute or the next one per http://www.gwg.nga.mil/ntb/baseline/docs/stdi0002/stdi0002_v20/4Mar1999.pdf

**Comment by derekwilhelm** [comment]
build now

**Comment by cxbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.codice.org/job/Alliance-Jobs/job/1.1.x/job/pr/job/Linux/21/<h2>Build result: FAILURE</span></h2>[...truncated 1.19 MB...][INFO] Total time: 04:12 min[INFO] Finished at: 2018-09-26T22:15:56+00:00[INFO] Final Memory: 66M/1075M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.owasp:dependency-check-maven:3.0.2:check (default) on project catalog-nsili-sourcestoquery-ui: One or more exceptions occurred during dependency-check analysis: One or more exceptions occurred during dependency-check analysis[ERROR] The download was interrupted; unable to complete the update[ERROR] -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionExceptionBuild step 'Invoke top-level Maven targets' marked build as failureSet build name.New build name is 'PR 645'Taking single-use slave mesos-jenkins-9589869f68b24f7daf6108425232fbe2-linux-medium offline.[BFA] Scanning build for known causes...[BFA] Exception during parsing file: java.nio.channels.ClosedByInterruptException[BFA] No failure causes found[BFA] Done. 8sAdding one-line test results to commit status...Setting status of ff043c5b5645d3db5a6c09a6b7a18d5eca99e287 to FAILURE with url https://jenkins.codice.org/job/Alliance-Jobs/job/1.1.x/job/pr/job/Linux/21/ and message: 'JOB FAILURE: https://jenkins.codice.org/job/Alliance-Jobs/job/1.1.x/job/pr/job/Linux/21/ 558 tests run, 0 skipped, 0 failed.'Using context: Internal CI Pipeline
:x: JOB FAILURE

**Comment by clockard** [comment]
build now

**Comment by cxbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.codice.org/job/Alliance-Jobs/job/1.1.x/job/pr/job/Linux/22/
:white_check_mark: JOB SUCCESS

**Comment by jlcsmith** [comment]
I wouldn't recommend addressing the codacy issues reported.  The code in question looks reasonable to me and within our coding conventions.

**Comment by ryeats** [comment]
I tested this out looks good 👍

**Comment by clockard** [comment]
build now

**Comment by cxbot1** [comment]
Internal build has been scheduled, your results will be available at build completion.

**Comment by cxbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.codice.org/job/Alliance-Jobs/job/1.1.x/job/pr/job/Linux/23/
:white_check_mark: JOB SUCCESS

**Pull request merged**